test/setup.ts

```diff
@@ -1,123 +1,314 @@
-    private_key,
-  } of (await hre.starknet.devnet.getPredeployedAccounts()).slice(0,2)) {
-    const account =
-      await hre.starknet.OpenZeppelinAccount.getAccountFromAddress(
-        address,
-        private_key
-      );
-    snPredeployedAccounts.push(account);
-  }
+  // preparation
+  const setupConfig = loadSnapshot();
+  console.log(setupConfig);
+  const mockStarknetMessaging = (
+    await hre.starknet.devnet.loadL1MessagingContract(hre.network.config.url!)
+  ).address as Address;
+
+  await initSnPredeployedAccounts(2);
 
-  const snVat = await deploySNVat(snPredeployedAccounts[0], snPredeployedAccounts[1].address)
+  const {
+    domains: { root: rootCfg, starknet: snCfg },
+  } = config;
 
-  const signers = await hre.ethers.getSigners();
-  const admin = await hre.ethers.getImpersonatedSigner(
-    "0xBE8E3e3618f7474F8cB1d074A26afFef007E98FB"
-  );
+  const [_, snOwner] = snPredeployedAccounts();
+
+  const deployer = (await hre.ethers.getSigners())[0];
+  const admin = await hre.ethers.getImpersonatedSigner(rootCfg.admin);
 
   // fund admin account
-  setBalance(admin.address, 10n ** 18n);
-
-  const deployer = signers[0];
-
-  // https://github.com/makerdao/dss-bridge/blob/4cfc84761b4bfeae747af14d3a2545377dd3304a/script/input/1/config.json
-
-  const daiJoin = await getDaiJoin(
-    "0x9759A6Ac90977b93B58547b4A71c78317f391A28"
-  );
-
-  const teleport = await dssTeleport.deploy(
-    deployer.address as Address,
-    admin.address as Address,
-    "0x48656c6c6f20576f726c64210000000000000000000000000000000000000000", // ilk
-    "0x48656c6c6f20576f726c64210000000000000000000000000000000000000000", // domain
-    "0x48656c6c6f20576f726c64210000000000000000000000000000000000000000", // hostDomain
-    daiJoin
-  );
-
-  const WAD = 10n ** 18n;
-  const _8_DAYS = 8n * 24n * 60n * 60n;
-
-  const fees: TeleportFees = await dssTeleport.deployLinearFee(
-    WAD / 10000n,
-    _8_DAYS
-  );
-
-  const dss: DssInstance = {
-    vat: await getVat("0x35D1b3F3D7966A1DFe207aa4514C12a259A0492B"),
-    jug: await getJug("0x19c0976f590D67707E62397C87829d896Dc0f1F1"),
-    cure: await getCure("0x0085c9feAb2335447E1F4DC9bf3593a8e28bdfc7"),
-    vow: await getVow("0xA950524441892A31ebddF91d3cEEFa04Bf454466"),
-  };
+  await setBalance(admin.address, "10");
 
-  const teleportConfig: DssTeleportConfig = {
-    debtCeiling: 10n ** 18n,
-    oracleThreshold: 5n,
-    oracleSigners: [],
-  };
+  // deploy on l1
+  console.log("getDss");
+  const dss = await getDss(rootCfg);
+
+  let teleport: TeleportInstance;
+  let snTeleport: SnTeleportInstance;
+  let snDss: SnDssInstance;
+  let snClaimToken: SnToken;
+  let fees: TeleportFees;
+  let snFee: SnTeleportConstantFee;
+  let hostAddress: Address;
+  let bridgeOracle: DSValue;
+  let host: DomainHost;
+  let guest: SnDomainGuest;
+
+  // If setupconfig is not an empty object, we skip deployments and get directly the contracts from this object
+  if (Object.keys(setupConfig).length !== 0) {
```

Comment:
Should be broken into at least to functions. Contract initilization should be invoked from mochaHooks.beforeEach